Tutorial: How to show a survey after a delay (#7081)
PostHog/posthog.com: contents/tutorials/delayed-survey.md

---
title: How to show a survey after a delay
date: 2023-11-01
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
tags: ['surveys']
---

> Before asking users about your app, you want them to **actually use it**. Surveys that pop up right away hinder their initial experience and get less useful responses as a result.

Reviewer: ```suggestion
Before you show a survey to your users, they need to _actually_ have a chance to use it. Surveys that pop up right away annoy users and get less useful responses as a result.
```

> To get the most accurate responses for your surveys, you want users to experience your site and product before asking. You want to wait and let them have their first impressions, instead of showing a survey right away.
> 
> To help you do this, you can delay your surveys and this tutorial shows you how. It goes over how to create a survey in PostHog and then implement a delay in your app for that survey.

Reviewer: ```suggestion
To get the most accurate responses for your surveys, you want users to experience your site and product before asking. You don't want to show your survey immediately. Instead, you should show your survey after a delay

This tutorial shows you how to do that. It goes over how to create a survey in PostHog and then implement a delay in your app for that survey.
```

> To help you do prevent this, you can set up a delay surveys and this tutorial shows you how. It goes over how to create a survey targeting a HTML class in PostHog and then implement a delay using that class and survey in your app.

Reviewer: Hmm this paragraph is still confusing. Its not clear why we target a survey with an HTML class in PostHog. We'd need to explain a bit on how PostHog works.

Maybe something like: "PostHog lets you target your survey based on HTML class. So our approach will show the class after a delay, so that...."

Also Im not sure the intro is the right place for this either. Feels like itll go into too much detail too quickly.

Author: Rewrote this and moved the class mention later

## Create your delayed survey

> First, we create our class-targeted survey in PostHog by going to the [Surveys tab](https://app.posthog.com/surveys) and clicking "New survey." Choose any template or create a blank survey, fill out the details, and open the "Targeting" section. 

Reviewer: "class targeted survey" is a confusing phrase. I think if you add the opening paragraph as mentioned above (explaining the approach), it will be easier to follow

```suggestion
First, we create our survey in PostHog by going to the [Surveys tab](https://app.posthog.com/surveys) and clicking "New survey." Choose any template or create a blank survey, fill out the details, and open the "Targeting" section. 
```

> In the targeting section, switch from "All users" to "Users who match…". Add `.delayed-survey` under "Selector matches" (the **leading period** is important) and click "Save as draft." Our implementation will add this class to the page when want the survey to show.

Reviewer: I think we're missing a paragraph describing what we'll do first. e.g. to create a survey with a delay, we'll target pages with a css class and then..."

This will make it easier to follow what we're doing

Author: Rewrote to try to make this clearer

> In the targeting section, start by switching from "All users" to "Users who match…". Under "Selector matches," add `.delayed-survey` (the **leading period** is important). This means our survey only shows when an HTML element with the class `delayed-survey` is on the page. Writing logic to add this class when we want the survey to display is the simplest way to set up a delay.

Reviewer: Not sure if the bold looks good or not. Worth a try?
```suggestion
In the targeting section, start by switching from "All users" to "Users who match…". Under "Selector matches," add `.delayed-survey` (the **leading period** is important). 

This means our survey **only shows when an HTML element with the class `delayed-survey` is on the page**. In the next step, we'll add code to only show this class after a delay. This way, our survey will show after a delay.
```

Finally, click "Save as draft" and then "Launch." You don’t have to worry about this survey showing early because the class selector won’t match anything until we implement it.

> ![Creating a survey video](../images/tutorials/delayed-survey/survey.mp4)

Reviewer: this video moves too fast and gave me a bit of vertigo 😅

## Implement your delayed survey

> Once installed in your app, PostHog automatically shows your surveys to targeted users. All we need to do is set up the logic to do the delay and add the class to our page.

Reviewer: ```suggestion
Once installed in your app, PostHog automatically shows your surveys to users when the `.delayed-survey` class is present. All we need to do is set up the logic to add this class to the page after a delay.
```

> Once installed in your app using the SDK or snippet, PostHog automatically shows your survey to targeted users. All we need to do is set up the logic to do the delay and add the class to our page.

Reviewer: Not sure why we mention SDK or snippet here all of sudden? We didnt mention it previously
